pg_shard.c

```diff
@@ -937,6 +973,12 @@ BuildDistributedPlan(Query *query, List *shardIntervalList)
 
 		deparse_shard_query(query, shardId, queryString);
 
+		if (LogDistributedStatements)
+		{
+			ereport(LOG, (errmsg("distributed statement: %s", queryString->data),
+						  errhidestmt(true)));
```

Comment:
BTW the config to emit the "main" query is `log_statement` and is orthogonal to this line.